### PR TITLE
[Utopia 560] add search box UI

### PR DESCRIPTION
**Pull request by binggaofw**
# Description

This PR includes the following proposed change(s):

- add search ui in pia table view and integrate with backend API

Screen Shot 2023-01-20 at 3 15 57 PM

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

**Review by kushal-arora-fw — COMMENTED** [review]
Local Testing Comments:
- Search box and search icon are misaligned
- Search icon is wider than in designs
- GET pia-intake list is getting called twice at the beginning
- Enter should submit and search
- "Search by title or drafter" is set as initial value instead of placeholder

File: src/frontend/src/pages/PIAListPage/index.tsx [hunk 1]
```diff
@@ -1,12 +1,30 @@
...
+import { faPlus, faMagnifyingGlass } from '@fortawesome/free-solid-svg-icons';
 import EmptyPIAList from '../../components/public/EmptyPIAList';
 import PIAListTable from '../../components/public/PIAListTable';
 import { usePIALookup } from '../../hooks/usePIALookup';
+import InputText from '../../components/common/InputText/InputText';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
 
 const PIAList = () => {
+  const history = useNavigate();
```
Comment (kushal-arora-fw) [inline src/frontend/src/pages/PIAListPage/index.tsx:11]: why calling useNavigate as history?

File: src/frontend/src/pages/PIAListPage/index.tsx [hunk 2]
```diff
@@ -1,12 +1,30 @@
...
   const tableHeadings = ['Title', 'Last modified', 'Drafter', 'PIA status'];
   const { tableData } = usePIALookup();
+
+  const [searchText, setSearchText] = useState('Search by title or drafter');
+  const handleClearSearchText = () => {
+    setSearchText('');
+    history('');
```
Comment (kushal-arora-fw) [inline src/frontend/src/pages/PIAListPage/index.tsx:18]: history('') - what does this mean?

File: src/frontend/src/pages/PIAListPage/index.tsx [hunk 3]
```diff
@@ -16,6 +34,34 @@ const PIAList = () => {
           <FontAwesomeIcon icon={faPlus} />
         </a>
       </div>
+
+      <div className="d-flex pt-3 ms-auto">
```
Comment (antsand) [inline src/frontend/src/pages/PIAListPage/index.tsx:38]: please add this as a search component in a different file.

**Review by antsand — CHANGES_REQUESTED** [review]

**Review by kushal-arora-fw — COMMENTED** [review]
Left few comments 

In addition to them,
If we are adding searchParam to url, then refreshing the page does not update the search field with the text

image

File: src/frontend/src/pages/PIAListPage/index.tsx (hunk 2 above)
Comment (binggaofw) [reply src/frontend/src/pages/PIAListPage/index.tsx:18]: bad naming, let's me change to navigate instead

File: src/frontend/src/pages/PIAListPage/index.tsx (hunk 1 above)
Comment (binggaofw) [reply src/frontend/src/pages/PIAListPage/index.tsx:11]: bad naming, change to navigate instead

**Review by kushal-arora-fw — COMMENTED** [review]
Comments unanswered:
- Search box and search icon are misaligned
- GET pia-intake list is getting called twice at the beginning
- Enter should submit and search
- Refresh page does not update search input value

**Comment by binggaofw** [comment]
@kushal-arora-fw  I know why it renders twice locally, it is the react feature. if you run the main branch in your local, you should see the two requests send to backend, but in dev env, you will only see one request be sent.

**Review by antsand — APPROVED** [review]

**Review by kushal-arora-fw — APPROVED** [review]
looks good

**Pull request merged**